include/exadg/incompressible_navier_stokes/spatial_discretization/operator_dual_splitting.cpp

```diff
@@ -613,15 +609,8 @@ OperatorDualSplitting<dim, Number>::local_interpolate_velocity_dirichlet_bc_boun
     if(boundary_type == BoundaryTypeU::Dirichlet or boundary_type == BoundaryTypeU::DirichletCached)
     {
       integrator.reinit(face);
-      integrator.read_dof_values(dst);
-
       for(unsigned int q = 0; q < integrator.n_q_points; ++q)
       {
-        unsigned int const local_face_number = matrix_free.get_face_info(face).interior_face_no;
-
-        unsigned int const index = matrix_free.get_shape_info(dof_index, quad_index)
-                                     .face_to_cell_index_nodal[local_face_number][q];
```

Comment:
In case we want to generalize rather this code towards simplex elements: We might want to take the class `ContainerInterfaceData` as a "template" to write a similar data structure for the mapping between q-points and degrees of freedom, see https://github.com/exadg/exadg/blob/5f70dfbcc5f8daac94e73859fb5d216dd49f8d76/include/exadg/functions_and_boundary_conditions/container_interface_data.h#L44-L58

Note that here in `ContainerInterfaceData`, the data type `MapVectorIndex` maps `Id = {face, q, v}` to the vector-index in the corresponding "data-arrays" of `ContainerInterfaceData`. Note that `dealii::types::global_dof_index` is not really used as a "dof-index" here.

For the present problem of mapping q-indices to dof-indices, maybe a similar data type, e.g. `MapDofIndices = std::map<Id, dealii::types::global_dof_index>` could be used for the mapping from q-indices to dof-indices? We could create this data structure in the setup phase and then read from that data structure during the simulation (very similar to the setup and use of `ContainerInterfaceData`).

We probably do not want to integrate this functionality into `ContainerInterfaceData`, because (i) not all use cases of `ContainerInterfaceData` need this information according to my understanding, and because (ii) we also need this information in cases where we do not have a corresponding `ContainerInterfaceData` (see function `local_interpolate_stress_bc_boundary_face()` below).